pkg/apis/v1alpha1/serviceexport.go

```diff
@@ -88,3 +92,133 @@ type ServiceExportList struct {
+	//
+	// * "NoService"
+	// * "InvalidServiceType"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionValid ServiceExportConditionType = "Valid"
+
+	// ServiceExportReasonValid is used with the "Valid" condition when the
+	// condition is True.
+	ServiceExportReasonValid ServiceExportConditionReason = "Valid"
+
+	// ServiceExportReasonNoService is used with the "Valid" condition when
+	// the associated Service does not exist.
+	ServiceExportReasonNoService ServiceExportConditionReason = "NoService"
+
+	// ServiceExportReasonInvalidServiceType is used with the "Valid"
+	// condition when the associated Service has an invalid type
+	// (per the KEP at least the ExternalName type).
+	ServiceExportReasonInvalidServiceType ServiceExportConditionReason = "InvalidServiceType"
+)
+
+const (
+	// ServiceExportConditionExported is true when the service is exported to some
+	// control plane or data plane.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "Exported"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "Invalid"
+	// * "Pending"
+	//
+	// Possible reasons for this condition to be Unknown are:
+	//
+	// * "Pending"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionExported ServiceExportConditionType = "Exported"
+
+	// ServiceExportReasonExported is used with the "Exported" condition
+	// when the condition is True.
+	ServiceExportReasonExported ServiceExportConditionReason = "Exported"
+
+	// ServiceExportReasonInvalid is used with the "Exported" condition
+	// when the exported service is invalid and will not be exported.
+	ServiceExportReasonInvalid ServiceExportConditionReason = "Invalid"
+
+	// ServiceExportReasonPending is used with the "Exported" condition
+	// when the service is going to be exported.
+	ServiceExportReasonPending ServiceExportConditionReason = "Pending"
+)
+
+const (
+	// ServiceExportConditionConflict indicates that the controller was unable
+	// to resolve conflict for a ServiceExport. This condition must be at
+	// least raised on the conflicting ServiceExport and is recommended to
+	// be raised on all on all the constituent `ServiceExport`s if feasible.
+	//
+	//
+	// Possible reasons for this condition to be true are:
+	//
+	// * "PortsConflict"
+	// * "TypeConflict"
+	// * "SessionAffinityConflict"
+	// * "SessionAffinityConfigConflict"
+	// * "MultipleConflicts"
+	//
+	// Possible reasons for this condition to be False are:
+	//
+	// * "NoConflicts"
+	//
+	// Controllers may raise this condition with other reasons,
+	// but should prefer to use the reasons listed above to improve
+	// interoperability.
+	ServiceExportConditionConflict ServiceExportConditionType = "Conflict"
+
```

Comment:
```suggestion

	// ServiceExportReasonConflicted is used with the "Conflicted"
	// condition when the exported service has multiple conflicts if the
	// implementation coalesces those or if an implementation-specific
	// conflict not matching one of the standard reasons is present.
	ServiceExportReasonConflicted ServiceExportConditionReason = "Conflicted"
	
```

Reply:
I applied @tpantelis suggestion to add commas let me know if you want to maintain this

Reply:
Might still be useful as a generic case, but no strong feelings.

Reply:
Does a project that have some very specific conflict should be using this generic reason or create their own though? If creating their own I would say that this shouldn't be necessary right?